src/utils/prices/prices.spec.ts

```diff
@@ -0,0 +1,716 @@
+              },
+            },
+          ],
+        },
+      ],
+    },
+  ],
+  others: [
+    {
+      exchange: 'UniswapV2',
+      srcAmount: '1000000000000000000',
+      destAmount: '3255989380',
+      unit: '3255989380',
+      data: {
+        router: '0x0000000000000000000000000000000000000000',
+        path: ['0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2', '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'],
+        factory: '0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f',
+        initCode: '0x96e8ac4277198ff8b6f785478aa9a39f403cb768dd02cbee326c3e7da348845f',
+        feeFactor: 10000,
+        pools: [
+          {
+            address: '0xB4e16d0168e52d35CaCD2c6185b44281Ec28C9Dc',
+            fee: 30,
+            direction: false,
+          },
+        ],
+        gasUSD: '13.227195',
+      },
+    },
+  ],
+  gasCostUSD: '10',
+  gasCost: '111111',
+  tokenTransferProxy: '0x3e7d31751347BAacf35945074a4a4A41581B2271',
+  contractAddress: '0x485D2446711E141D2C8a94bC24BeaA5d5A110D74',
+  contractMethod: 'swapOnUniswap',
+  srcUSD: '5000',
+  destUSD: '5000',
+  partner: 'paraswap.io',
+  partnerFee: 0,
+  maxImpactReached: false,
+  hmac: '319c5cf83098a07aeebb11bed6310db51311201f',
+}
+
+const ZEROX_PRICE_QUOTE_RESPONSE = {
+  price: '0.123456789101112',
+  estimatedPriceImpact: '0',
+  value: '0',
+  gasPrice: '42000000000',
+  gas: '111000',
+  estimatedGas: '111000',
+  protocolFee: '0',
+  minimumProtocolFee: '0',
+  buyTokenAddress: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+  buyAmount: '1234567890',
+  sellAmount: '123456789',
+  sellTokenAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
+  sources: [
+    {
+      name: 'Uniswap',
+      proportion: '0',
+    },
+  ],
+  allowanceTarget: '0xdef1c0ded9bec7f1a1670819833240f027b25eff',
+  sellTokenToEthRate: '1',
+  buyTokenToEthRate: '10',
+}
+
+beforeEach(() => {
+  fetchMock.resetMocks()
+})
+
+afterEach(() => {
+  jest.restoreAllMocks()
+})
+
+describe('getAllPricesLegacy: Common logic', () => {
+  // GIVEN
+  const chainId = SupportedChainId.MAINNET
+  const BUY_QUOTE_PARAMS = {
+    chainId,
+    fromDecimals: 18,
+    toDecimals: 18,
+    baseToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+    quoteToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+    amount: '1234567890',
+    kind: OrderKind.BUY,
+    validTo: Date.now() + 30000,
+  }
+  const ZRX_BUY_QUOTE_RESPONSE = { ...ZEROX_PRICE_QUOTE_RESPONSE, chainId }
+  const PARA_BUY_QUOTE_RESPONSE = { ...PARASWAP_PRICE_QUOTE_RESPONSE, side: SwapSide.BUY }
+
+  test('Returns CoW price and null for uninstantiated APIs', async () => {
+    const cowSdk = new CowSdk(chainId, { signer }) // no additional APIs instantiated
+    // mocks cow/0x/para quote response
+    _mockApiResponses({
+      cow: {
+        response: COW_PRICE_QUOTE_RESPONSE,
+        responseOptions: GOOD_SERVER_RESPONSE,
+      },
+      para: { response: null },
+      zrx: { response: null, responseOptions: GOOD_SERVER_RESPONSE },
```

Comment:
You missed a few spots where the api is not instantiated by `responseOptions` are set

Reply:
👍